### ISSUE TITLE
BE: Add reusable Q&As to CLP

### ISSUE THREAD

**Issue by jilladams** [department-of-veterans-affairs/va.gov-cms#10938]
### User story

**AS A** Content Editor
**I WANT** to add reusable Q&As to Campaign Landing Pages (CLPs)
**SO THAT** Q&As have a single source of truth. 

### Engineering notes / background

Follow the runbook created in #10936. 

Designs are in [Figma](https://www.figma.com/file/MTBtolRaZK4hUkCprXyIiD/Q%26As?type=design&node-id=4%3A2157&mode=design&t=J07jlBp0p9rkSrAl-1) – the CLP tab/workspace.

FAQ section exists on CLPs currently and will continue to be present (since not all FAQs will / should be reusable).

The [KB article about CLPs](https://prod.cms.va.gov/help/campaign-landing-pages/how-to-manage-campaign-landing-pages) notes that one can add FAQs to CLPs, so that will need to be modified. Eng on this ticket should capture enough notes to provide guidance for the KB.

<details>

![Screenshot 2023-07-19 at 11 23 01 AM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/85581471/2ffabdc4-8b7b-4837-9d85-53ca5fab6b7a)
</details>

Assumption:
- New naming conventions already applied in prereq ticket #15098.

## Acceptance criteria

- [x] Reusable Q&As can be added to CLP nodes **in draft PR**
  - [x] but DON'T MERGE UNTIL the FE is ready
- [x] Design, section names, and help text follow new Figma design
  - [x] Reusable Q&A section
  - [x] Page-specific Q&A section
  - [x] Includes indent/theming of page-specific Q&A UI
- [x] Updated [runbook](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/content/tier-2-content-IA-and-design/cope-cms-faq-library/engineering/README.md) with any changes discovered while following it
- [x] A11y review

**Comment by jilladams** [comment]
@laflannery FYI this ticket will put Q&As onto CLPs, and we'd like your eyes on that while in Tugboat. (It won't merge to Prod during this sprint.)

**Comment by laflannery** [comment]
@jilladams I have a question about this - Q&As have a heading issue on the accordions. Is there space to fix that here or is the scope of this ticket just copying existing implementation?

**Comment by jilladams** [comment]
Yes that's right - this is just adding the option for Reusable Q&A to CLPs. Addressing the page-specific issue would need a sep ticket.

**Comment by chri5tia** [comment]
# Status Update 10/5/23

As of just before scrum this morning, this work is mostly complete but needs testing. There is a piece I need to troubleshoot. If I haven't already, the next step is to push the PR up in order to potentially collaborate on the issue, described below, but I my goal is to have it resolved and the PR ready to review by scrum tomorrow.

## Roadblock details
[Screenshot coming]

"Enable this page segment" for FAQ is unchecked but the new reusable Q&A shows up, and is unaffected by the checkbox. I need to refresh how this feature was built and where it is in the code so I can see what is needed in order to add the new form field to that functionality.

**Comment by chri5tia** [comment]
# Status update 10/6/23
PR is ready for code review and Design and PO sign off
PR will not be merged in this sprint until FE is ready

**Comment by jilladams** [comment]
@chri5tia moved this back to "Review / QA" column until the PR is merged. (Still in draft, here: https://github.com/department-of-veterans-affairs/va.gov-cms/pull/15577)

**Comment by chri5tia** [comment]
# Status Update 10/9/23

Work is completed and PR is ready to merge but _not merged yet_ according to ACs.

## Notes for Future work

CLP node needs a description.

Consider moving `_va_gov_backend_get_clp_toggles()` to the `va_gov_clp` module.

**Comment by chri5tia** [comment]
@laflannery Can I get a11y sign off on this work? You can view the changes here: https://cms-ulclxmkmxt58l5utev56h9griwv3v1cp.ci.cms.va.gov/

**Comment by chri5tia** [comment]
@jilladams I didn't need to make any changes to the run book. I pinged Laura for review. I will ask Jordan about the indentation. I don't recall what that is.

**Comment by chri5tia** [comment]
## Status update 10/11/23

Per Laura's observations, I am syncing up with Jordan shortly to ensure nothing was missed and ensure that Drupal and Figma are in sync. Scheduled this for 10/12.

Moving this ticket back into "in progress".

**Comment by chri5tia** [comment]
- [ ] Change "Place Q&As" to "Add Reusable Q&As"

**Comment by thejordanwood** [comment]
@laflannery Thanks for looking this over so thoroughly! My Figma document is actually a bit out of date on some of this. 

1. Christia has made this change (yay!)
2. Figma doc is out of date. This was one of the first designs before I realized that all reusable Q&As should be groups. What Christia has is correct and I'll update Figma to show this. 
3. Figma doc is out of date. 
4. Christia is going to update the copy in the modal. 
5. Figma doc is out of date and Christia's design is correct. We discovered recently that some Q&As have multiple answers, so I don't think the design can be changed too much on this. I'll update Figma to show this.
